src/coreclr/src/vm/appdomain.cpp

```diff
@@ -6243,6 +6247,8 @@ HRESULT RuntimeInvokeHostAssemblyResolver(INT_PTR pManagedAssemblyLoadContextToB
     {
         bool fResolvedAssembly = false;
 
+        tracer.SetManagedALC(pManagedAssemblyLoadContextToBindWithin);
```

Comment:
Why not set this right where we set the name - the ALC is available there as well.